src/stdlib/mod.rs

```diff
@@ -147,13 +162,42 @@ pub struct FunctionResult {
     pub signature: String,
 }
 
+fn default_arg_insert_text(arg: &str, index: usize) -> String {
+    (format!("{}: ${}", arg, index + 1))
+}
+
+async fn arg_insert_text(
+    arg: &str,
+    index: usize,
+    ctx: RequestContext,
+) -> String {
+    match arg {
+        "bucket" => {
+            if let Ok(buckets) = ctx.callbacks.get_buckets().await {
```

Comment:
This is a little challenging to read because of how nested it is. Since only one path has a different return value, I wonder if it can be rewritten to be more easily understandable.

Reply:
since this is an if let to extract the value of buckets from Result I am not sure this can be avoided.

Reply:
Yeah I wasn't sure either :(

Reply:
I went ahead created a function get_bucket_insert_text to break this up a bit